extension/js/background_page/migrations.ts

```diff
@@ -26,3 +40,60 @@ export const migrateGlobal = async () => {
     console.info('done migrating');
   }
 };
+
+export const moveContactsToEmailsAndPubkeys = async (db: IDBDatabase): Promise<void> => {
+  if (!db.objectStoreNames.contains('contacts')) {
+    return;
+  }
+  console.info('migrating contacts of ContactStore to emails and pubkeys...');
+  const batchSize = 50;
+  while (await moveContactsBatchToEmailsAndPubkeys(db, batchSize)) {
```

Comment:
I think it's ok as it is. If it crashes, and half of contacts don't get converted, it will try again next time I restart browser.

The only change would be to ensure that the plugin can start up and continue functioning if this method throws.

Reply:
I wrapped the call in `try/catch` and the extension is usable after conversion failure now.

Reply:
I tested it. The records beyond and including the failed batch are retained in `contacts`. They can't be found by the app but will be retried on the next background start.